azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/ChatMiddleware.kt

```diff
@@ -10,29 +10,39 @@ import com.azure.android.communication.ui.chat.redux.action.Action
 import com.azure.android.communication.ui.chat.redux.action.ChatAction
 import com.azure.android.communication.ui.chat.redux.action.ErrorAction
 import com.azure.android.communication.ui.chat.redux.state.ReduxState
+import com.azure.android.communication.ui.chat.service.ChatService
+import com.azure.android.communication.ui.chat.utilities.CoroutineContextProvider
 
 internal interface ChatMiddleware
 
+// ChatMiddleware
+//
+// Manages
+// ChatServiceListener (Service -> Redux)
+// ChatActionHandler (Redux -> Service)
 internal class ChatMiddlewareImpl(
-    private val chatActionHandler: ChatActionHandler,
-    private val chatServiceListener: ChatServiceListener
+    chatService: ChatService,
+    coroutineContextProvider: CoroutineContextProvider
 ) :
     Middleware<ReduxState>,
     ChatMiddleware {
+
+    private val chatServiceListener = ChatServiceListener(chatService = chatService, coroutineContextProvider = coroutineContextProvider)
+    private val chatActionHandler = ChatActionHandler(chatService =  chatService)
```

Comment:
Adam, this may create issue for unit testing? I think every dependency should be injected.